base/shared/src/main/scala/scalaz/data/ConstInstances.scala

```diff
@@ -21,30 +20,54 @@ trait ConstInstances {
     override def toList[A](fa: Const[R, A]): List[A] = Nil
   }
 
-  implicit def constFunctor[R]: Functor[Const[R, ?]] = new Functor[Const[R, ?]] {
+  trait ConstFunctor[R] extends Functor[Const[R, ?]] {
```

Comment:
There remains, unless I've missed it, an outer pointer per inner trait. Not sure how much of a difference that makes.

Reply:
I don't see how.